blobfuse/src/BlobfuseGcCache.cpp

```diff
@@ -164,6 +164,19 @@ void gc_cache::run_gc_cache()
             {
                 //clean up the file from cache
                 int fd = open(mntPath, O_WRONLY);
+                if (errno == 13) {
+                    close(fd);
+                    int fdr = open(mntPath, O_RDONLY);
+                    if (fdr > 0) {
+                        errno = 0;
+                        if (fchmod(fdr, config_options.defaultPermission) == 0) {
+                            AZS_DEBUGLOGV("GCClean clould not open for delete, so, Write protection overwritten for %s, resubmitting for cleanup", mntPath);
+                            }
+                        //TODO: If there is lock issue below convert permissions back to what it was.
+                        close(fdr);
+                    }
```

Comment:
If readonly also fails then we can dump a log saying this file will not get evicted from cache due to the error.

Reply:
done